### PR TITLE
Bug fix for Import-PSGetRepository in Windows PS

### LINKED ISSUE #1442: Import-PSGetRepository fails silently to import UNC based PowerShellGet v2 repositories on PowerShell 5.1
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://www.powershellgallery.com/packages/Microsoft.PowerShell.PSResourceGet)
- [X] Search the existing issues.

### Steps to reproduce

`Import-PSGetRepository` fails silently to import **UNC** based  PowerShellGet v2 repositories on **PowerShell 5.1**
*(It works on PowerShell 7.3.8)*

Steps to reproduce:
1. Register an UNC PowerShellGet v2 repository (remote PC requried) on PowerShell 5.1:
```powershell
Register-PSRepository -Name Test -SourceLocation \\Server\RepoPath -InstallationPolicy Trusted
```
2.  Try to import the PowerShellGet v2 repository:
```powershell
Import-PSGetRepository -Verbose -Force
````
3. The UNC repository is not imported:
```powershell
Get-PSResourceRepository
```

### Expected behavior

```console
PS C:\> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.3.8
PSEdition                      Core
GitCommitId                    7.3.8
OS                             Microsoft Windows 10.0.19045
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

PS C:\> Register-PSRepository -Name PowerEdge -SourceLocation \\poweredge\transfer -InstallationPolicy Trusted
Package source with Name: PowerEdge added successfully.
PS C:\> Get-PSRepository

Name                      InstallationPolicy   SourceLocation
----                      ------------------   --------------
PowerEdge                 Trusted              \\poweredge\transfer
PSGallery                 Trusted              https://www.powershellgallery.com/api/v2

PS C:\> Import-PSGetRepository -Verbose -Force
VERBOSE: Found 2 registered PowerShellGet repositories.
VERBOSE: Selected 1 NuGet repositories.
VERBOSE: Registering PowerEdge at \\poweredge\transfer -Trusted:$True -Force:$True.

Name      Uri                       Trusted Priority
----      ---                       ------- --------
PowerEdge file://poweredge/transfer True    50

PS C:\> Get-PSResourceRepository -Verbose

Name      Uri                                      Trusted Priority
----      ---                                      ------- --------
PowerEdge file://poweredge/transfer                True    50
PSGallery https://www.powershellgallery.com/api/v2 False   50

PS C:\>
```

### Actual behavior

```console
PS C:\> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.19041.3031
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.19041.3031
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1

PS C:\> Register-PSRepository -Name PowerEdge -SourceLocation \\poweredge\transfer -InstallationPolicy Trusted
Package source with Name: PowerEdge added successfully.
PS C:\> Get-PSRepository

Name                      InstallationPolicy   SourceLocation
----                      ------------------   --------------
PSGallery                 Trusted              https://www.powershellgallery.com/api/v2
PowerEdge                 Trusted              \\poweredge\transfer


PS C:\> Import-PSGetRepository -Verbose -Force
VERBOSE: Found 2 registered PowerShellGet repositories.
VERBOSE: Selected  NuGet repositories.
PS C:\> Get-PSResourceRepository -Verbose

Name      Uri                                      Trusted Priority
----      ---                                      ------- --------
PSGallery https://www.powershellgallery.com/api/v2 False   50

PS C:\>
```

### Error details

```console
No error
```

### Environment data

```powershell
ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Binary     1.0.0      Microsoft.PowerShell.PSResourceGet  {Find-PSResource, Get-InstalledPSResource, Get-PSResourceRepository, Get-PSScriptFileInfo...}



Name                           Value
----                           -----
PSVersion                      5.1.19041.3031
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.19041.3031
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

### Visuals

_No response_

### PR DESCRIPTION
**Pull request by alerickson**
# PR Summary
Bug fix for `Import-PSGetRepository` in Windows PowerShell not actually porting PowerShellGet v2 repositories to PSResourceGet.  `.count` property does not work on the object returned in Windows PS (however it does in PS 6+).  Simple change to check if the `$repos` returned is empty or not checks to see whether repositories need to be transferred over. 

## PR Context
Resolves #1442 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

### CODE REVIEW

File: src/Microsoft.PowerShell.PSResourceGet.psm1 [hunk 1]
```diff
@@ -31,7 +31,7 @@ function Import-PSGetRepository {
 
         Microsoft.PowerShell.Utility\Write-Verbose ('Selected {0} NuGet repositories.' -f $repos.Count)
```
Comment (kborowinski) [inline src/Microsoft.PowerShell.PSResourceGet.psm1:34]: @alerickson: The issue is also here, as `$repos.Count` will return `$null` on PowerShell 5.1, when only one repository was found, and therefore the verbose message will be misleading - *"VERBOSE: Selected  NuGet repositories"* .

I suggest changing it either to `@($repos).Count`, or maybe, to make the code cleaner, we should assign the `repos` count to a variable, like this;
```powershell
...
$reposCount = @(repos).Count
Microsoft.PowerShell.Utility\Write-Verbose ('Selected {0} NuGet repositories.' -f $reposCount)

if ($reposCount) {
...
```
or just convert the filtering result to an array:
```powershell
...
$repos = @(
    $PSGetRepositories.Values |
        Microsoft.PowerShell.Core\Where-Object {$_.PackageManagementProvider -eq 'NuGet'-and $_.Name -ne 'PSGallery'} |
        Microsoft.PowerShell.Utility\Select-Object Name, Trusted, SourceLocation
)

Microsoft.PowerShell.Utility\Write-Verbose ('Selected {0} NuGet repositories.' -f $repos.Count)

if ($repos.Count) {
...
```

Comment (alerickson) [reply src/Microsoft.PowerShell.PSResourceGet.psm1:34]: Thanks for catching that, let me fix that

File: src/Microsoft.PowerShell.PSResourceGet.psm1 [hunk 2]
```diff
@@ -25,13 +25,14 @@ function Import-PSGetRepository {
     Microsoft.PowerShell.Utility\Write-Verbose ('Found {0} registered PowerShellGet repositories.' -f $PSGetRepositories.Count)
 
     if ($PSGetRepositories.Count) {
-        $repos = $PSGetRepositories.Values |
+        $repos = @( $PSGetRepositories.Values |
```
Comment (kborowinski) [inline src/Microsoft.PowerShell.PSResourceGet.psm1]: Sorry @alerickson for nitpicking, but do we need extra space here? And wouldn't it look better if the `$PSGetRepositories.Values` got moved to a new line instead:
```powershell
$repos = @(
    $PSGetRepositories.Values |
```

File: src/Microsoft.PowerShell.PSResourceGet.psm1 (hunk 1 above)
Comment (kborowinski) [reply src/Microsoft.PowerShell.PSResourceGet.psm1:34]: LGTM! Thanks!

**Review by kborowinski — APPROVED** [review]
LGTM! Thanks! 🎉

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**